src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1169,11 +1175,17 @@ def set_up_unbind_service_bindings(self):
                 new_template["serviceBinds"] = existing_template.get("serviceBinds", [])
 
             service_bindings_dict = {}
+            java_component_name_set = {}
```

Comment:
ni:
Please add comment about if resource is java component type, with auto change the `-` in binding name to `_`.